index.html

```diff
@@ -1831,8 +1831,92 @@ <h3>Verify Proof</h3>
       <section class="normative">
         <h3>Retrieve Verification Method</h3>
 
-      </section>
+        <p>
+The following algorithm specifies how to safely retrieve a verification method,
+such as a cryptographic public key, by using a <a>verification method</a>
+identifier contained in a <a>data integrity proof</a>. Required inputs are a
+<a>data integrity proof</a> (<var>proof</var>) and a set of
+<strong>dereferencing options</strong> (<var>options</var>). A
+<strong>verification method</strong> is produced as output.
+        </p>
+
+        <ol class="algorithm">
+          <li>
+Let <var>vmIdentifier</var> be set to
+<var>proof</var>.<var>verificationMethod</var>.
+          </li>
+          <li>
+Let <var>vmPurpose</var> be set to <var>proof</var>.<var>proofPurpose</var>.
+          </li>
+          <li>
+If <var>vmIdentifier</var> is not a valid URL, an
+`INVALID_VERIFICATION_METHOD_URL` error MUST be raised.
+          </li>
+          <li>
+Let <var>controllerDocumentUrl</var> be the result of parsing
+<var>vmIdentifier</var> according to the rules of the URL scheme and extracting
+the primary resource identifier (without the fragment identifier).
+          </li>
+          <li>
+Let <var>vmFragment</var> be the result of parsing <var>vmIdentifier</var>
+according to the rules of the URL scheme and extracting the secondary
+resource identifier (the fragment identifier).
+          </li>
+          <li>
+Let <var>controllerDocument</var> be the result of dereferencing
+<var>controllerDocumentUrl</var>, according to the rules
+of the URL scheme and using the supplied <var>options</var>.
+          </li>
+          <li>
+If <var>controllerDocument</var>.<var>id</var> does not match the
+<var>controllerDocumentUrl</var>, a `INVALID_CONTROLLER_DOCUMENT_ID` error MUST
+be raised.
+          </li>
+          <li>
+If <var>controllerDocument</var> is not a valid <a>controller document</a>,
+an `INVALID_CONTROLLER_DOCUMENT` error MUST be raised.
+          </li>
+          <li>
+Let <var>verificationMethod</var> be the result of dereferencing the
+<var>vmFragment</var> from the <var>controllerDocument</var> according to the
+rules of the associated URL scheme and using the supplied <var>options</var>.
+          </li>
+          <li>
+If <var>verificationMethod</var> is not a valid <a>verification method</a>,
+an `INVALID_VERIFICATION_METHOD` error MUST be raised.
+          </li>
+          <li>
+If <var>verificationMethod</var> is not associated with the array of
+<var>vmPurpose</var>s in the <var>controllerDocument</var>, either by reference
```

Comment:
Something else might also need to be changed for ReSpec to properly handle this plural.
```suggestion
<var>vmPurposes</var> in the <var>controllerDocument</var>, either by reference
```